### PR TITLE
Consistent testing

### PR DESCRIPTION
**Pull request by Liamdoult**
Convert the existing GPU classes into one standard format and extend GPU testing support.

### CODE REVIEW

File: test/test_mnist.py [hunk 1]
```diff
@@ -22,9 +22,9 @@ def fetch_mnist():
 
 # create a model
 class TinyBobNet:
```
Comment (geohot) [inline test/test_mnist.py:24]: Instead of adding GPU here, can you use the helper to convert it after you create it?

File: tinygrad/tensor.py [hunk 2]
```diff
@@ -87,20 +87,20 @@ def shape(self):
     return self.data.shape
 
   @staticmethod
-  def zeros(*shape):
-    return Tensor(np.zeros(shape, dtype=np.float32))
+  def zeros(*shape, gpu=False):
```
Comment (geohot) [inline tinygrad/tensor.py:90]: These are good

**Comment by geohot** [comment]
Still very supportive of the extended testing, and added the gpu kwarg to the constructors like in this PR. Clean up and I'll merge?

**Comment by Liamdoult** [comment]
Hey, I will resolve this tomorrow!

File: test/test_mnist.py (hunk 1 above)
Comment (Liamdoult) [reply test/test_mnist.py:24]: I am not sure what you want here.

If by helper function you mean `.cpu` and `.gpu`,  it seems non-trivial to apply the functions for each of the `Tensor` parameters inside the models.

If you just don't want the GPU kwarg, perhaps I can create a base class that has an auto gpu-cpu conversion for the model?

Perhaps I am missing something obvious?

Comment (geohot) [reply test/test_mnist.py:24]:
```
[x.cuda_() for x in get_parameters(model)]
```

File: test/test_tensor.py [hunk 3]
```diff
@@ -55,14 +57,26 @@ def test_gradcheck(self):
...
 
     # coarse approx. since a "big" eps and the non-linearities of the model
     self.assertFalse(gradcheck(tiny_func, tiny_x, eps = 0.1))
 
+
+@unittest.skipUnless(GPU, "Requires GPU")
+class TestTinygradGPU(TestTinygrad):
+  gpu = True
+
+  @unittest.skip("float32 not supported on GPU")
```
Comment (geohot) [inline test/test_tensor.py:74]: Hmm, float32 is actually the only thing supported on GPU. Is this comment wrong?

File: test/test_mnist.py (hunk 1 above)
Comment (Liamdoult) [reply test/test_mnist.py:24]: This should be resolved now.

The optimizers also create Tensor objects on initialization. This means that all the optimizer parameters needed to be converted. To account for this, I updated the get_parameters function.

File: test/test_tensor.py (hunk 3 above)
Comment (Liamdoult) [reply test/test_tensor.py]: Yep, float64*. Resolved.

**Comment by geohot** [comment]
Nice refactor of get_parameters too!

**Pull request merged**